### PR TITLE
refactored code, down load from github instead of kaggle

### LINKED ISSUE #59: making the CI more reliable
We have a bunch of integration tests that fetch data and notebooks using Kaggle's API. However, they fail sometimes (probably because the Kaggle API isn't very reliable). See https://github.com/ploomber/soorgeon/issues/58

One solution is to use the Kaggle API once and then upload the files to an S3 bucket, then have the CI download them from S3 instead of Kaggle.

### PR DESCRIPTION
**Pull request by Wxl19980214**
## Describe your changes
1. Instead of fetching from kaggle, we fetch from github by pygithub api now
2. Add _pygithub.py for downloading a dir from a repo.
3. Add pygithub package to setup.py (I don't know if this one is necessary)
5. Removed stored notebooks under _kaggle since we can store them in the storage repo now
6. Removed unused functions under src/_kaggle.py
7. Removed kaggle username and key in CI, I don't think we need them. Maybe we can also remove them from repo secrets
8.  Storage repo [here](https://github.com/ploomber/ci-notebooks) and README.md has instructions on how to upload
9. Contribute.md remains unchanged, we can let user register new notebook in _kaggle/index.yaml. But the rest of downloading and uploading, updating CI and test cases will need to be done manually by one of the team members.

## To do
1.We need to add a personal access token to the repo's secrets.
2. Right now CI will fail because this one fails test_notebooks[look-at-this-note-feature-engineering-is-easy] for this reason:
E               soorgeon.exceptions.InputError: Only H1 headings are found. At this time, only H2 headings are supported. Check out our guide: 
## Issue ticket number and link
Closes #59 #39 

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have added thorough tests (when necessary).
- [x] I have added the right documentation (when needed). Product update? If yes, write one line about this update.

### CODE REVIEW

**Comment by Wxl19980214** [comment]
@edublancas  Yeah it is. I put it in the To Do list. We need a PAT in repo's secrets. Also one of the test case is failing anyway

**Comment by edublancas** [comment]
ok so most of them pass now. but there's one that still fails `look-at-this-note-feature-engineering-is-easy`. did anything change from the file we had here to the one that's being downloaded? looks like it has H1 headings now

**Comment by idomic** [comment]
Yeah we're on it, we introduced a feature to reject H1s, we'll change it to H2s in the notebook and that should fix it.

**Comment by Wxl19980214** [comment]
Just updated notebooks. The last check should pass. But still, I do think we need to put a token in to make sure it does't fail in the future. I think there's a limitation on api request for non-login request. I am asking shuyang to generate a pat to test if it works. I don't even know why it works without authentication, but it mostly because our repo is public.

**Comment by idomic** [comment]
lmk and I'll remove the PAT. Yeah I think it's because it's a public repo.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -60,8 +60,7 @@ jobs:
           pip install -r _kaggle/requirements.lock.txt
       - name: Integration tests
         env:
-          KAGGLE_KEY: ${{ secrets.KAGGLE_KEY }}
-          KAGGLE_USERNAME: ${{ secrets.KAGGLE_USERNAME }}
+          PERSONAL_ACCESS_TOKEN: ${{ secrets.PERSONAL_ACCESS_TOKEN}}
```
Comment (idomic) [inline .github/workflows/ci.yaml:63]: Can we remove this if we're no longer using this secret?

File: src/soorgeon/_pygithub.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+from github import Github
+# import os
+
+
+def download_directory(dir):
+    """
+    Download all contents from a dir in a repository,
+    store data in a sub_dir called input
+    """
+    # token = os.getenv("PERSONAL_ACCESS_TOKEN")
```
Comment (idomic) [inline src/soorgeon/_pygithub.py:13]: Same here, remove if not used.

**Review by idomic — CHANGES_REQUESTED** [review]
Added small comments

**Comment by Wxl19980214** [comment]
@idomic I left them there on purpose just in case we encounter some errors later. But I can remove it for now. It's easy fix anyway

**Pull request merged**